### PR TITLE
Various docs edits

### PR DESCRIPTION
**Pull request by dmontagu**
In response to this comment https://github.com/samuelcolvin/pydantic/pull/856#issuecomment-538341962.

I tried to put special emphasis into ensuring consistent use of conventions within each section of the docs, and across the docs on the whole.

If you notice any patterns you don't like, let me know and I'll change it everywhere (no need for you to go through them repeatedly marking the same thing one-by-one).

Feel free to reject this PR if there are a number of changes you don't like, but you also don't want to spend the time reviewing it.

------

This isn't comprehensive, and isn't intended to be (yet) -- there are some files I haven't gotten to, and I'm sure there are some things I've still missed even in the ones I did review. But I figured it would be smarter to submit changes incrementally in case you didn't like some/all of them.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -53,27 +53,27 @@ outputs:
...
+: *pydantic* data structures are just instances of classes you define, so auto-completion, linting,
+  [mypy](usage/mypy.md), IDEs (especially [PyCharm](pycharm_plugin.md)), and your intuition should 
   all work properly with your validated data.
 
 **dual use**
-: *pydantic's* [BaseSettings](usage/settings.md) class allows it to be used in both a "validate this request data"
-  context and "load my system settings" context. The main difference being that system settings can have defaults
-  changed by environment variables and more complex objects like DSNs and python objects are often required.
+: *pydantic's* [BaseSettings](usage/settings.md) class allows *pydantic* to be used in both a "validate this request
+  data" context, and it a "load my system settings" context. The main difference is that system settings can read
```
Comment (samuelcolvin) [inline docs/index.md]:
```suggestion
  data" context, and in a "load my system settings" context. The main difference is that system settings can read
```

File: docs/index.md [hunk 2]
```diff
@@ -53,27 +53,27 @@ outputs:
...
+  defaults from environment variables, and more complex objects like DSNs and python objects are often required.
```
Comment (samuelcolvin) [inline docs/index.md]: after an extended discussion in the office we think there shouldn't be a comment here.

File: docs/index.md [hunk 3]
```diff
@@ -53,27 +53,27 @@ outputs:
...
 
 **fast**
-: In [benchmarks](benchmarks.md) *pydantic* is faster than all other tested libraries.
+: In [benchmarks](benchmarks.md), *pydantic* is faster than all other tested libraries.
```
Comment (samuelcolvin) [inline docs/index.md]: or here.

File: docs/usage/dataclasses.md [hunk 4]
```diff
@@ -9,26 +9,27 @@ Dataclasses work in python 3.6 using the [dataclasses backport package](https://
 _(This script is complete, it should run "as is")_
 
 !!! note
-    Keep in mind that `pydantic.dataclasses.dataclass` is a drop in replacement for `dataclasses.dataclass`
-    with validation, not a replacement for `pydantic.BaseModel`. There are cases where subclassing
+    Keep in mind that `pydantic.dataclasses.dataclass` is a drop-in replacement for `dataclasses.dataclass`
+    with validation, *not* a replacement for `pydantic.BaseModel`. There are cases where subclassing
```
Comment (samuelcolvin) [inline docs/usage/dataclasses.md]:
```suggestion
    with validation, **not** a replacement for `pydantic.BaseModel`. There are cases where subclassing
```
Maybe?

File: docs/usage/model_config.md [hunk 5]
```diff
@@ -3,42 +3,41 @@ Behaviour of pydantic can be controlled via the `Config` class on a model.
 Options:
 
 **`title`**
-: title for the generated JSON Schema
+: the title for the generated JSON Schema
 
 **`anystr_strip_whitespace`**
-: strip or not trailing and leading whitespace for str & byte types (default: `False`)
+: whether to strip leadin and trailing whitespace for str & byte types (default: `False`)
```
Comment (samuelcolvin) [inline docs/usage/model_config.md]:
```suggestion
: whether to strip leading and trailing whitespace for str & byte types (default: `False`)
```

File: docs/index.md [hunk 6]
```diff
@@ -93,25 +93,25 @@ Hundreds of organisations and packages are using *pydantic*, including:
   [for subprojects](https://github.com/samuelcolvin/pydantic/issues/773).
 
 **Microsoft**
-: are using *pydantic* (via FastAPI) for 
-  [numerous services](https://github.com/tiangolo/fastapi/pull/26#issuecomment-463768795) some of which are 
+: is using *pydantic* (via FastAPI) for
```
Comment (samuelcolvin) [inline docs/index.md]: it's debatable, bit I think "are" reads better here.

**Comment by samuelcolvin** [comment]
Sorry, I've made a few changes that conflict. Also I've now merged the new docs so you might need to rebase to master.

**Comment by samuelcolvin** [comment]
no problem, no great hurry. I might release v1.0b2 tonight to check that the markdown stuff works correctly with pypi, but this can come later.

File: docs/index.md (hunk 2 above)
Comment (dmontagu) [reply docs/index.md]: I'm not sure whether you were referring to both commas or just the first one. Either way, I think you're right -- I tend to do commas by intuition based on how I am inclined to read the sentence aloud, but thinking more logically here I see they aren't necessary.

I'm inclined to keep the second comma though since I find it hard to parse the sentence otherwise. (It's not clear that the second part is distinct from the first until you reach the end; I find the comma helps me realize this as I go.)

Also, I think it should probably be changed from saying "The main difference is" to "The main differences are" given there are two 😄.

I'll rework it tonight.

File: docs/index.md (hunk 3 above)
Comment (dmontagu) [reply docs/index.md]: agreed

File: docs/index.md (hunk 6 above)
Comment (dmontagu) [reply docs/index.md]: Looks like this is a bit of an Anglicism -- https://english.stackexchange.com/questions/133105/organisation-singular-or-plural/133106#133106. Happy to keep it that way for the euro flair -- I'll rework them all to be consistent later.

**Comment by dmontagu** [comment]
Hmm, I'm going to make a new PR against master.
